pkg/chunk/encoding/bigchunk.go

```diff
@@ -281,7 +281,7 @@ func (it *bigchunkIterator) Scan() bool {
 
 	for it.i < len(it.chunks)-1 {
 		it.i++
-		it.curr = it.chunks[it.i].Iterator()
+		it.curr = it.chunks[it.i].Iterator(nil)
```

Comment:
```suggestion
		it.curr = it.chunks[it.i].Iterator(&it.curr)
```

Reply:
I had another PR open to take care of many iterator re-use https://github.com/cortexproject/cortex/pull/1574, hence I did not include any in this PR. Would you like to add it in this PR itself?

Reply:
Will add this part in this PR itself